src/anndata/compat/__init__.py

```diff
@@ -46,7 +47,7 @@ class Empty:
     pass
 
 
-Index1D = slice | int | str | np.int64 | np.ndarray
+Index1D = slice | int | str | np.int64 | np.ndarray | EllipsisType
 Index = Index1D | tuple[Index1D, Index1D] | scipy.sparse.spmatrix | SpArray
```

Comment:
Since we use `Index1D` elsewhere, I think we should use more correct types:

`...` doesn‘t stand for one dimension, it stands for “the other dimensions”, so it only makes sense in the context of a multidimensional array

```suggestion
Index1D = slice | int | str | np.int64 | np.ndarray
IndexRest = Index1D | Ellipsis
Index = IndexRest | tuple[Index1D, IndexRest] | tuple [IndexRest, Index1D] | scipy.sparse.spmatrix | SpArray
```

Reply:
ah, sorry, I mean [types.EllipsisType](https://docs.python.org/3/library/types.html#types.EllipsisType)

I thought it would work like `None`, but I guess since `... is Ellipsis` and `...` has special meaning for typing, they needed to distinguish `tuple[int, ...]` (int tuple of unspecified length) from `tuple[int, EllipsisType]` (duple with an int and an `Ellipsis`/`...`)

Reply:
Similarly, the two-elem tuple conflicts with https://github.com/scverse/anndata/pull/1729#discussion_r1810772405